students/tim_lurvey/lesson02/assignment/code/charges_calc.py

```diff
@@ -0,0 +1,220 @@
+"""
+Returns total price paid for individual rentals
+"""
+import argparse
+import json
+import datetime
+import math
+import logging
+
+def parse_cmd_arguments():
+    """
+    Parse arguments from command line and return values
+    """
+    debug_help_cmd = "Debugging level" \
+                     "\n0: No debug messages or log file." \
+                     "\n1: Only error messages." \
+                     "\n2: Error messages and warnings." \
+                     "\n3: Error messages, warnings and debug messages."
+
+    parser = argparse.ArgumentParser(description='Process some integers.',
+                                     formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.add_argument('-i', '--input',
+                        help='input JSON file',
+                        required=True)
+
+    parser.add_argument('-o', '--output',
+                        help='ouput JSON file',
+                        required=True)
+
+    parser.add_argument('-d', '--debug',
+                        help=debug_help_cmd,
+                        required=False,
+                        default='0')
+
+    return parser.parse_args()
+
+
+def set_logging_level():
```

Comment:
For this one, try to consider the possibility of the user selecting an invalid logging level. Currently, the program crashes, which is fine (i.e., the fact that the user entered an invalid parameter will be noticed) but could be better handled using exceptions:

```
def set_logging_level():
    """
    Determine the logging level from the command line and set terminal and file logging.
    Default is no log file or debug terminal printing
    """
    log_level = {'0': {'term': logging.NOTSET,
                    'file': logging.NOTSET},
                '1': {'term': logging.ERROR,
                    'file': logging.ERROR},
                '2': {'term': logging.WARNING,
                    'file': logging.WARNING},
                '3': {'term': logging.DEBUG,
                    'file': logging.WARNING},
                }

    format_str = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"

    try:
        log_terminal = logging.StreamHandler()
        log_terminal.setFormatter(fmt=logging.Formatter(format_str))
        log_terminal.setLevel(level=log_level.get(args.debug).get('term'))

        log_file = logging.FileHandler(filename=datetime.datetime.now().strftime("%Y-%m-%d") + ".log")
        log_file.setFormatter(fmt=logging.Formatter(format_str))
        log_file.setLevel(level=log_level.get(args.debug).get('file'))
    except AttributeError:
        print("Invalid logging level selected")
        exit()

    logger = logging.getLogger()
    logger.setLevel(logging.DEBUG)
    logger.addHandler(log_terminal)
    logger.addHandler(log_file)
```